### DISCUSSION TITLE
How to solve the scenario of slow building of a  large number of pages

### DISCUSSION THREAD

**Discussion by sedationh** [vuepress/core#765] [Q&A] [answered]
As the title suggests, I have created a number of markdown files using scripts.
They are close to thousand in number, and have the same format and contain some images.
When in dev mode, it works well. But when i build it, it is too slow that i cant use vercel to deploy this.
How to solve this problem?
![image](https://user-images.githubusercontent.com/56129212/159199375-effc3e39-3341-4209-b4b0-b02042de7ce7.png)

<img width="386" alt="image" src="https://user-images.githubusercontent.com/56129212/159199368-011c79b5-ce75-422e-b7e2-123c0fae672e.png">

**Comment by Mister-Hope** [comment, marked as answer]
Any static site generator has proper range of your content, if you have too many content, e.g.(100 times comparing vuepress official docs), a static docs generator should not be a good chosie for you.

You are not actually describing how many content are you trying to build with (e.g.: 1000 pages with average 3000 words per page).  IMO, your project's word number should not beyond millon level . If so,a static site generator which requires you to built is denitely not what you want,  try tools with backend and properly based on database, as they do not require a build stage, but dynamic renders content when users visite a page.


Also:

1. for my blog, webpack bundler is faster than vite. You may try switching it if it helps.
2. You may try to build in another places (e.g. GitHub workflow limit is 2 hours), and then delopy to vercel manully.

**Reply by sedationh** [reply]
Can I render the page dynamically?
For example, I have a data.json file and use route path to match the information in it.

Also, what does 
> try those tools with backend support

mean specifically?

**Reply by Mister-Hope** [reply]
Original answers updated.

**Reply by Mister-Hope** [reply]
> Can I render the page dynamically?

You may achieve that by writing components, a VuePress App is acutally a highly customized Vue App, you can do most thing you can do when using vue-cli with it.
